test/cloud_build_github_gke_test.go

```diff
@@ -0,0 +1,143 @@
+		githubRepo := "sample-app-docker"
+		gkeClusterTerratestOptions := createTestGitHubTerraformOptions(t, uniqueID, project, region, githubOrg, githubRepo, cloudBuildGitHubGkeTerraformModulePath)
+		test_structure.SaveString(t, workingDir, "uniqueID", uniqueID)
+		test_structure.SaveString(t, workingDir, "project", project)
+		test_structure.SaveString(t, workingDir, "region", region)
+		test_structure.SaveTerraformOptions(t, workingDir, gkeClusterTerratestOptions)
+		test_structure.SaveKubectlOptions(t, workingDir, kubectlOptions)
+	})
+
+	defer test_structure.RunTestStage(t, "cleanup", func() {
+		project := test_structure.LoadString(t, workingDir, "project")
+		buildID := test_structure.LoadString(t, workingDir, "buildID")
+
+		build := gcp.GetBuild(t, project, buildID)
+		for _, image := range build.GetImages() {
+			gcp.DeleteGCRRepo(t, image)
+		}
+
+		gkeClusterTerratestOptions := test_structure.LoadTerraformOptions(t, workingDir)
+		terraform.Destroy(t, gkeClusterTerratestOptions)
+
+		kubectlOptions := test_structure.LoadKubectlOptions(t, workingDir)
+		err := os.Remove(kubectlOptions.ConfigPath)
+		require.NoError(t, err)
+	})
+
+	test_structure.RunTestStage(t, "terraform_apply", func() {
+		gkeClusterTerratestOptions := test_structure.LoadTerraformOptions(t, workingDir)
+		terraform.InitAndApply(t, gkeClusterTerratestOptions)
+	})
+
+	test_structure.RunTestStage(t, "configure_kubectl", func() {
+		gkeClusterTerratestOptions := test_structure.LoadTerraformOptions(t, workingDir)
+		kubectlOptions := test_structure.LoadKubectlOptions(t, workingDir)
+		project := test_structure.LoadString(t, workingDir, "project")
+		region := test_structure.LoadString(t, workingDir, "region")
+		clusterName := gkeClusterTerratestOptions.Vars["cluster_name"].(string)
+
+		// gcloud beta container clusters get-credentials example-cluster --region australia-southeast1 --project dev-sandbox-123456
+		cmd := shell.Command{
+			Command: "gcloud",
+			Args:    []string{"beta", "container", "clusters", "get-credentials", clusterName, "--region", region, "--project", project},
+			Env: map[string]string{
+				"KUBECONFIG": kubectlOptions.ConfigPath,
+			},
+		}
+
+		shell.RunCommand(t, cmd)
+	})
+
+	test_structure.RunTestStage(t, "wait_for_workers", func() {
+		kubectlOptions := test_structure.LoadKubectlOptions(t, workingDir)
+		verifyGkeNodesAreReady(t, kubectlOptions)
+	})
+
+	test_structure.RunTestStage(t, "trigger_build", func() {
+		// write a test file
+		date := []byte(fmt.Sprintf("%s\n", time.Now().String()))
+		testFile := fmt.Sprintf("%s/auto-committed.txt", os.Getenv("SAMPLE_APP_DIR"))
+		err := ioutil.WriteFile(testFile, date, 0644)
+		if err != nil {
+			t.Fatalf("Could not write temporary file")
+		}
+
+		// commit and push
+		cmd := shell.Command{
+			Command:    "git-add-commit-push",
+			Args:       []string{"--remote-name", "origin", "--path", testFile, "--message", "triggering a build", "--skip-git-pull", "--skip-ci-flag", ""},
```

Comment:
Is there a way this can be made to work using a custom branch that is deleted? I worry about endlessly growing the repo with a stream of test commits to the sample apps on the main line.